internal/pkg/cli/app_list.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+	"github.com/spf13/cobra"
+)
+
+const (
+	applicationListProjectNamePrompt = "Which project's applications would you like to list?"
+	applicationListProjectNameHelper = "A project groups all of your applications together."
+)
+
+// ListAppOpts contains the fields to collect for listing an application.
+type ListAppOpts struct {
+	ShouldOutputJSON    bool
+	ShouldShowLocalApps bool
+
+	applications  []*archer.Application
+	manager       archer.ApplicationLister
+	projectGetter archer.ProjectGetter
+	projectLister archer.ProjectLister
+
+	ws archer.Workspace
+	w  io.Writer
+
+	*GlobalOpts
+}
+
+func (opts *ListAppOpts) selectProject() (string, error) {
+	projs, err := opts.projectLister.ListProjects()
+	if err != nil {
+		return "", err
+	}
+	var projStrs []string
+	for _, projStr := range projs {
+		projStrs = append(projStrs, projStr.Name)
+	}
+	if len(projStrs) == 0 {
+		log.Infoln("There are no projects to select.")
+		return "", nil
+	}
+	proj, err := opts.prompt.SelectOne(
+		applicationListProjectNamePrompt,
+		applicationListProjectNameHelper,
+		projStrs,
+	)
+	return proj, err
+}
+
+func (opts *ListAppOpts) localAppsFilter(appNames []string) {
+	var localApps []*archer.Application
+	for _, appName := range appNames {
+		for _, app := range opts.applications {
+			if appName == app.Name {
+				localApps = append(localApps, app)
+			}
+		}
+	}
+	opts.applications = localApps
+}
+
+// Ask asks for fields that are required but not passed in.
+func (opts *ListAppOpts) Ask() error {
+	if opts.ProjectName() != "" {
+		return nil
+	}
+	projectName, err := opts.selectProject()
+	if err != nil {
+		return fmt.Errorf("failed to get project name: %w", err)
+	}
+	opts.projectName = projectName
+
+	return nil
+}
+
+// Execute lists the applications through the prompt.
+func (opts *ListAppOpts) Execute() error {
+	// Ensure the project actually exists before we try to list its applications.
+	if _, err := opts.projectGetter.GetProject(opts.ProjectName()); err != nil {
+		return err
+	}
+
+	apps, err := opts.manager.ListApplications(opts.ProjectName())
+	if err != nil {
+		return err
+	}
+	opts.applications = apps
+
+	if opts.ShouldShowLocalApps {
+		localAppManifests, err := opts.ws.Apps()
+		if err != nil {
+			return fmt.Errorf("failed to get local app manifests: %w", err)
+		}
+		var localAppNames []string
+		for _, appManifest := range localAppManifests {
+			localAppNames = append(localAppNames, appManifest.AppName())
+		}
+		opts.localAppsFilter(localAppNames)
+	}
+
+	var out string
+	if opts.ShouldOutputJSON {
+		data, err := opts.jsonOutput()
+		if err != nil {
+			return err
+		}
+		out = data
+	} else {
+		out = opts.humanOutput()
+	}
+	fmt.Fprintf(opts.w, out)
+
+	return nil
+}
+
+func (opts *ListAppOpts) humanOutput() string {
+	b := &strings.Builder{}
+	for _, app := range opts.applications {
+		fmt.Fprintf(b, "%s: %s\n", app.Type, app.Name)
+	}
+	return b.String()
```

Comment:
Should we remove the ":" to be consistent with the style guide (https://github.com/aws/amazon-ecs-cli-v2/blob/master/STYLE_GUIDE.md#data)

I think that's the dream look that we want to achieve with our human outputs :)